internal/services/aadgraph/application_app_role_resource.go

```diff
@@ -128,9 +129,9 @@ func applicationAppRoleResourceCreateUpdate(d *schema.ResourceData, meta interfa
 	app, err := client.Get(ctx, id.ObjectId)
 	if err != nil {
 		if utils.ResponseWasNotFound(app.Response) {
-			return fmt.Errorf("Application with ID %q was not found", id.ObjectId)
+			return tf.ErrorDiag(fmt.Sprintf("Application with object ID %q was not found", id.ObjectId), "", "application_object_id")
```

Comment:
And these could become something like
```suggestion
			return tf.ErrorDiagPathF(nil, "application_object_id", "Application with object ID %q was not found", id.ObjectId))
```